pyvrp/cpp/bindings.h

```diff
@@ -69,10 +69,18 @@ template <pyvrp::MeasureType T> struct type_caster<pyvrp::Measure<T>>
 {
     PYBIND11_TYPE_CASTER(pyvrp::Measure<T>, _("int"));
 
-    bool load(pybind11::handle src,
-              [[maybe_unused]] bool convert)  // Python -> C++
+    bool load(pybind11::handle src, bool convert)  // Python -> C++
     {
-        auto const raw = PyLong_AsLong(src.ptr());
+        if (!convert && !PyLong_Check(src.ptr()))  // only int when conversion
+            return false;                          // is not allowed.
+
+        PyObject *tmp = PyNumber_Long(src.ptr());  // any argument for which
+        if (!tmp)                                  // Python's int() succeeds.
+            return false;
```

Comment:
A consequence of this is that floating point data is all cast to ints, because
```python
x = 3.5
y = int(x)
```
is perfectly fine Python code. As this is in Python and is also how e.g. OR-Tools does it, I don't think we need to raise for floating-point inputs.